src/gc/gcdiag.h

```diff
@@ -0,0 +1,26 @@
+#ifndef __GCDIAG_H__
+#define __GCDIAG_H__
+
+typedef void (* record_surv_fn)(uint8_t*, uint8_t*, ptrdiff_t, size_t, BOOL, BOOL);
+typedef void (* fq_walk_fn)(BOOL, void*);
+
+class GCDiagnostics
+{
+public:
+    static void BeforeGCStart(size_t index, int gen, int reason);
+    static void GCStart(int gen, bool isInduced);
+    static void SegmentCreation(size_t start, size_t size, int type);
+    static void UpdateGenerationBounds();
+    static void GCEnd(size_t index, int gen, int reason, bool fConcurrent);
+
+    static void WalkFReachableObjects(gc_heap* hp);
```

Comment:
(Maybe I do not understand what kind of separation is this change is trying to achieve.)